lib/src/serialization/deserializer.dart

```diff
@@ -0,0 +1,64 @@
+// Copyright (C) 2024 Intel Corporation
+// SPDX-License-Identifier: BSD-3-Clause
+//
+// deserializer.dart
+// A deserialization block, deserializing narrow input data onto a wide channel.
+//
+// 2024 August 27
+// Author: desmond Kirkpatrick <desmond.a.kirkpatrick@intel.com>
+
+import 'package:collection/collection.dart';
+import 'package:rohd/rohd.dart';
+import 'package:rohd_hcl/rohd_hcl.dart';
+
+/// [Deserializer] aggregates data from a serialized stream.
+class Deserializer extends Module {
+  /// Aggregated data output.
+  LogicArray get deserialized => output('deserialized') as LogicArray;
+
+  /// Length of aggregate to deserialize.
+  int get length => _length;
+
+  /// [done] emitted when the last element is committed to [deserialized].
+  /// The timing is that you can latch [deserialized] when [done] is high.
+  Logic get done => output('done');
+
+  /// Return the current count of elements that have been serialized out.
+  Logic get count => output('count');
+
+  /// Internal storage of length of data to be serialized.
+  late final int _length;
+
+  /// Build a Deserializer that takes serialized input [serialized]
+  /// and aggregates it into one wide output [deserialized] of length [length].
+  ///
+  /// Updates one element per clock while [enable] (if connected) is high,
+  /// emitting [done] when completing the filling of wide output `LogicArray`
+  /// [deserialized].
+  Deserializer(Logic serialized, this._length,
+      {required Logic clk,
+      required Logic reset,
+      Logic? enable,
+      super.name = 'Deserializer'}) {
```

Comment:
the `name` is used as a default instance name, so typically we write it in lower-case.  As written, this could be a little confusing for generated verilog since `Deserializer Deserializer(.clk(...` is actually legal SystemVerilog (same name for instance and definition).  I recommend changing default to `deserializer` in lower-case.  (same for the serializer)